### ISSUE TITLE
Issuer listing page changes for Coming soon state

### ISSUE THREAD

**Issue by andywalton** [Neufund/platform-frontend#2884]
### Background

We need to allow our Issuers to list their ETO campaigns earlier, currently our process requires them to have their Investment Terms, Term Sheet & Nominee all setup before we can list them. We want to change this so that we can list them early and say Coming soon and show their public company page with only marketing information completed.

Please refer to the epic Issuer States Update for more detail background: #2879 

Because we're making changes to the Issuer flow, this early listing means we need to make some changes to our Issuer listing / company page as they won't have certain information available at the time we list them. Fields like Term sheet, Investment terms etc. won't be available. We need to disable, hide and display TBA (to be announced) for some fields. The tasks below describe this.

### Ticket Tasks

**Tasks 1-3**

![Screenshot 2019-05-08 at 15 12 01](https://user-images.githubusercontent.com/609657/57377819-a7c7ae00-71a3-11e9-8c5f-27b0f93b0dda.png)

Using the above image as a reference, these 3 changes are required:

- [x] 1. When we list an Issuer publicly as Coming soon, an Investor should see the state `Coming soon` as the status for this ETO. This should be reflected on the ETO preview card as well.

- [x] 2. In this public state (ie. Coming soon)all three tags should be disabled (grayed out) and inactive as these documents aren't yet available to the public. The unavailable documents are: Term sheet; Investment memorandum and Smart contract on chain.

- [x] 3. Because the Key Investment Terms values won't be available. Make sure the following fields display the acronym: TBA (to be announced)

**Key Investment Terms:**
Pre-Money Valuation: TBA
Investment Amount: TBA
New shares generated: TBA
Equity token price: TBA

---

**Task 4**

![Screenshot 2019-05-08 at 15 13 32](https://user-images.githubusercontent.com/609657/57377973-04c36400-71a4-11e9-9bdc-349c44fde172.png)

Scroll down until you get to ETO Timeline and make the following change:

- [x] 4. ETO timeline change for the Whitelist label

Whilst an Issuer is published with only marketing data on the timeline we should show should say “Coming Soon” instead of “Whitelist” in the first phase.

And then this label should change to “Campaign” once the ETO is submitted with Investment and Term sheet information.

Only once the user enables the Whitelist the label should change to “Whitelist”.

---

**Task 5**

![Screenshot 2019-05-08 at 15 13 36](https://user-images.githubusercontent.com/609657/57378498-2ffa8300-71a5-11e9-926e-d2f962e9adf6.png)

Scroll down until you get to ETO Investment Terms and make the following change:

- [x] 5.1) Don't hide this section, show TBA for all numbers and values.
- [x] 5.2) Hid the document download links - ie all the agreement downloads

---

**Task 6**

![Screenshot 2019-05-08 at 15 13 41](https://user-images.githubusercontent.com/609657/57378647-849dfe00-71a5-11e9-946d-ae83208a7e54.png)

Scroll down until you get to Documents and make the following change:

- [x] 6. We hide all legal documents and in this section only show marketing documents in this state

---

### Technical notes
Please see 
**Task 1 and 4**
that is basically new logic for naming statuses of ETO and corresponds to ticket https://github.com/neufund/platform-frontend/issues/2786
1. when in `preview` and `pending` states when `is_marketing_data_visible_in_preview` is `visible` we name the state `Coming Soon`
2. as above but `is_marketing_data_visible_in_preview` is **not** `visible` we name the state "Eto Application"
3. in `listed` and `prospectus_approved` and `onchain` states when `is_bookbuilding_enabled` is true we name the state "Whitelisting", if false: `Campaigning` (so the old logic of displaying Campaigning stays, we just add new condition for `is_bookbuilding_enabled`)

In the ETO timeline if we are before PRE-SALE we just display current value of the state as defined above. so it reflects state on the card

**Other Tasks**
You do not need to do any if-else stuff. Backend will remove all investment terms fields and you simply need to handle this accordingly. In particular
- documents and templates collection will be empty so lists and green tags will be disabled automatically
- all the fields you map in terms will be missing from `eto_data`

Please note that this applies only to `eto_data` that you get via public methods (where you provide preview code directly or in the search method), all the `me` methods will work as usual so for example
https://platform.neufund.io/eto/view/ - will return all the data as you set them up in the forms
https://platform.neufund.io/eto/view/3862e0eb-931e-416d-854f-5c5da7dd2731 - will remove investmnt terms in `preview` and `pending` states

### Testing notes
We'll need a fixture that allows us to test an Issuer company that is in this specific Coming soon state.

**Comment by rudolfix** [comment]
@desfero please mark tasks that are already done.

here's the state matrix, actual state names may change but that we'll handle ourselves
https://docs.google.com/spreadsheets/d/1ES_KQFF2hLpj1A0ANnwuYEJCtMnFTYquLZABuKvtAsw/edit#gid=138831629

tab: state Matrix - this table maps form backend and smart contract statuses and flags into names of the states and colors

would be cool if this is easy to change, for example you define some general naming rules and then exceptions... but if just writing code with ifs is easier then go for it. idk :)

one more explanation is necessary: we also provide different state names when ISSUER is looking at this HOWEVER this is only display in `me` context so
- in dashboard
![image](https://user-images.githubusercontent.com/17202864/57652893-e5c43800-75d0-11e9-86d3-c288515dc46b.png)

- in profile preview but only when seen as issuer

**Comment by rudolfix** [comment]
comment on `is_bookbuilding_enabled` - this may be `true` in LISTING, PROSPECTUS APPROVED and ONCHAIN states in `eto_data`, currently you switch new investment cards to "WHITELISTING" where we show pledges only in ONCHAIN. it should be visible (and status set accordingly) in all states above.

our old investment widget works exactly this way

**Comment by glupiawymowka** [comment]
*on .io* 
 - [x] timeline changes accordingly to the state 
- [x] inactive document tags 
- [x] legal docs section hidden 
- [x] TBA appears accordingly 


to check later: 
- [ ] TBA underline/tooltip
- [ ] ETO status shown

**Comment by glupiawymowka** [comment]
tested on stage
